### DISCUSSION TITLE
Prisma error when adding `resolveInput` hook on relationship

### DISCUSSION THREAD

**Discussion by killercup** [keystonejs/keystone#8596] [Questions] [answered]
**Summary:** I wanted to add a hook that sets creator of an entity to the currently logged in user. I will create this with the API, not the admin UI.

## Steps to reproduce the problem

I started from the auth example.
Here's a simplified excerpt from my config, hope it still replicates.

```
export const lists: Lists = {
    User: list({
        fields: {
            email: text({ isIndexed: 'unique' }),
            password: password(),
        },
    }),
    Presentation: list({
        fields: {
            name: text({ validation: { isRequired: true } }),
            creator: relationship({
                ref: "User",
                hooks: {
                    resolveInput: async ({ resolvedData, context, inputData, operation, item }) => {
						// there was logic here, but breaks with just this:
                        return resolvedData;
                    },
                }
            }),
        },
    })
};
```

## What I expected to happen

Works as before

## Error

```
Invalid `model.update()` invocation in
$PWD/node_modules/@keystone-6/core/dist/createSystem-2c6190fa.cjs.dev.js:935:97
[…]
Unknown arg `creator` in data.creator for type PresentationUncheckedUpdateInput. Did you mean `creatorId`? Available args: […]
```

## What works instead

Moving the hook to the list type

**Comment by dcousens** [comment]
@killercup what is invoking this? (how are you calling `update`)

**Comment by killercup** [comment]
This is from me pressing save in the admin UI. Sorry, my goal was to use the API for this directly, but I was still trying to get it to work at this point.

(Since the workaround of just using a hook at the list level works, I haven't dug any deeper into this.)

**Comment by dcousens** [comment, marked as answer]
Oh, sorry, I missed the problem anyway.
`resolvedData` is not a field, it is the resolved item.

You need to only return the field value, `resolvedData[fieldKey]` (`fieldKey` is passed to `resolveInput`)
I too think `resolvedData` not mapping to the field value here, is confusing.

**Comment by killercup** [comment]
Ohh, thanks! It's a bit confusing that it still type checks.
